org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/compiler/OnTheFlyJavaCompiler2.java

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) 2015, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.xbase.compiler;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map;
+
+import org.eclipse.jdt.core.compiler.CategorizedProblem;
+import org.eclipse.xtend2.lib.StringConcatenation;
+import org.eclipse.xtext.util.JavaVersion;
+import org.eclipse.xtext.xbase.lib.Conversions;
+import org.eclipse.xtext.xbase.lib.Exceptions;
+import org.eclipse.xtext.xbase.lib.Functions.Function0;
+import org.eclipse.xtext.xbase.lib.IterableExtensions;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Iterables;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+/**
+ * @author Sven Efftinge - Initial contribution and API
+ * @noextend This class is not intended to be subclassed by clients.
+ * @noreference This class is not intended to be referenced by clients.
+ * 
+ * @since 2.9
+ * @deprecated Use org.eclipse.xtext.xbase.testing.OnTheFlyJavaCompiler2 instead
+ */
+@Deprecated
+@Singleton
+public class OnTheFlyJavaCompiler2 {
+	private InMemoryJavaCompiler inMemoryCompiler;
+
+	/**
+	 * Creates a new OnTheFlyCompiler that accepts Java6 compliant code.
+	 */
+	@Inject
+	public OnTheFlyJavaCompiler2(ClassLoader scope) {
+		this(scope, JavaVersion.JAVA6);
+	}
+
+	public OnTheFlyJavaCompiler2(ClassLoader scope, JavaVersion version) {
+		this.inMemoryCompiler = new InMemoryJavaCompiler(scope, version);
+	}
+
+	public Class<?> compileToClass( String classname,  String code) {
+		InMemoryJavaCompiler.Result result = inMemoryCompiler.compile(new JavaSource(toJavaFile(classname), code));
+		try {
+			if (IterableExtensions.exists(result.getCompilationProblems(), CategorizedProblem::isError)) {
+				StringConcatenation richString = new StringConcatenation();
+				richString.append("Java code compiled with errors:");
+				richString.newLine();
+				richString.append(Joiner.on("\n")
+						.join(IterableExtensions.filter(result.getCompilationProblems(), CategorizedProblem::isError)));
+				richString.newLineIfNotEmpty();
+				richString.newLine();
+				richString.append("Code was:");
+				richString.newLine();
+				richString.append(code);
+				richString.newLineIfNotEmpty();
+				throw new IllegalArgumentException(richString.toString());
+			}
+			return result.getClassLoader().loadClass(classname);
+		} catch (Throwable t) {
+			if (t instanceof ClassNotFoundException) {
+				ClassNotFoundException e = (ClassNotFoundException) t;
```

Comment:
This is just an error string ... maybe just use ordinary string concatenation!?

Reply:
dont want spend to much time refactoring deprecated code